src/test/java/io/aiven/connect/elasticsearch/ElasticsearchSinkFailureTest.java

```diff
@@ -0,0 +1,192 @@
+                                                        final JestResultHandler<? super T> jestResultHandler) {
+            jestResultHandler.completed((T) success);
+        }
+
+        @Override
+        public void shutdownClient() {
+
+        }
+
+        @Override
+        public void setServers(final Set<String> servers) {
+
+        }
+    }
+
+    abstract class CustomJestElasticsearchClient extends JestElasticsearchClient {
+        public final AtomicInteger apiCallCounter = new AtomicInteger(0);
+
+        public CustomJestElasticsearchClient(final JestClient client) {
+            super(client);
+        }
+
+        @Override
+        public Version getVersion() {
+            return Version.ES_V7;
+        }
+
+        @Override
+        public JsonObject getMapping(final String index, final String type) {
+            return new JsonObject();
+        }
+    }
+
+    class CustomJestTemporarilyFailingElasticsearchClient extends CustomJestElasticsearchClient {
+        public CustomJestTemporarilyFailingElasticsearchClient(final JestClient client) {
+            super(client);
+        }
+
+        @Override
+        public BulkResponse executeBulk(final BulkRequest bulk) throws IOException {
+            final int numAttempt = apiCallCounter.incrementAndGet();
+
+            if (numAttempt < 3) {
+                return BulkResponse.failure(true, "");
+            }
+
+            return BulkResponse.success();
+        }
+    }
+
+    class CustomJestFailingElasticsearchClient extends CustomJestElasticsearchClient {
+        public CustomJestFailingElasticsearchClient(final JestClient client) {
+            super(client);
+        }
+
+        @Override
+        public BulkResponse executeBulk(final BulkRequest bulk) throws IOException {
+            apiCallCounter.incrementAndGet();
+            return BulkResponse.failure(false, "");
+        }
+    }
+
+    @Test
+    public void testRetryIfRecoverable() {
+        final BulkResult success = new BulkResult(new Gson());
+        success.setSucceeded(true);
+
+
+        final ElasticsearchSinkTask elasticsearchSinkTask = new ElasticsearchSinkTask();
+        final CustomJestElasticsearchClient failingClient =
+            new CustomJestTemporarilyFailingElasticsearchClient(new MockedJestClient(success));
+
+        final Map<String, String> props = getConfigWithFilteringOfInvalidMessages();
+
+        elasticsearchSinkTask.start(props, failingClient);
+        elasticsearchSinkTask.put(Collections.singletonList(new SinkRecord("topic", 0, null, null, null, "foo", 0)));
+        elasticsearchSinkTask.flush(Collections.emptyMap());
+        elasticsearchSinkTask.put(Collections.singletonList(new SinkRecord("topic", 0, null, null, null, "bar", 0)));
+        elasticsearchSinkTask.flush(Collections.emptyMap());
+        // 3 retries + 1 success
+        assert failingClient.apiCallCounter.get() == 4;
```

Comment:
nit: I'd suggest using assertj as it's used in more up to date connector plugins, but it's not added as dependency here. So, at least let's use junit assertions as it's a more conventional way of doing things in our java projects: 
```suggestion
        Assert.assertEquals(4, failingClient.apiCallCounter.get());
```
ps. consider static import for Assert.assertEquals method(s).